crates/bitwarden-crypto/src/keys/device_key.rs

```diff
@@ -63,8 +63,8 @@ impl DeviceKey {
         let device_private_key: Vec<u8> = protected_device_private_key.decrypt_with_key(&self.0)?;
         let device_private_key = AsymmetricCryptoKey::from_der(device_private_key.as_slice())?;
 
-        let dec: Vec<u8> = protected_user_key.decrypt_with_key(&device_private_key)?;
-        let user_key: SymmetricCryptoKey = dec.as_slice().try_into()?;
+        let mut dec: Vec<u8> = protected_user_key.decrypt_with_key(&device_private_key)?;
+        let user_key: SymmetricCryptoKey = dec.as_mut_slice().try_into()?;
```

Comment:
I don't believe this consumes the key. Do we need to zero `dec`?

Reply:
We zeroize the key inside the `TryFrom<&mut [u8]>` impl:

https://github.com/bitwarden/sdk/blob/6e844469c9e1555a5abc45248ea88abcc025679b/crates/bitwarden-crypto/src/keys/symmetric_crypto_key.rs#L90-L115

Reply:
That feels potentially confusing? Do you expect a `TryFrom` of a slice to manipulate the input? I do see the benefit of ensuring it gets consumed though.

Reply:
We could also just pass an owned Vec<> and consume it in the function, but that will limit the use of the API for someone trying to pass a GenericArray or a [u8; 64] for example.

I think the fact that the function receives a `&mut [u8]` should make it clear enough that something is being modified, but we could add a comment mentioning explicitly that it's being zeroed.

Reply:
Let's add a comment to make it explicit. We can revaluate if this turns out to be confusing, potentially not using try from and instead having an explicit function could make it less confusing.